### PR TITLE
Update notes on allowed naming convention for tables

### PR DESCRIPTION
**Pull request by jflayhart**
I created some tables with hyphens and they threw the following errors:

image

Changing to use only `_` instead of `-` fixed it.

### CODE REVIEW

**Review by Jackie-Jiang — COMMENTED** [review]
Good catch. Actually `-` is allowed in table name, but it is a reserved key in SQL (represent minus), so in order to query the table, the table name has to be double quoted

File: configuration-reference/table.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ description: The tables below shows the properties available to set at the table
 
 | Property         | Description                                                                                                                                                                                                                                                                      |
 | ---------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
-| tableName        | Specifies the name of the table. Should only contain alpha-numeric characters, hyphens (‘-‘), or underscores (‘\__’). (Using a double-underscore (‘\_\__’) is not allowed as it is reserved for other features within Pinot)                                                     |
+| tableName        | Specifies the name of the table. Should only contain alpha-numeric characters and/or underscores (‘\__’). (Using a double-underscore (‘\_\__’) is not allowed as it is reserved for other features within Pinot)                                                     |
```
Comment (matthewhelmke) [inline configuration-reference/table.md]:
```suggestion
| tableName        | Specifies the name of the table. Should only contain alpha-numeric characters, hyphens (‘-‘), or underscores (‘\__’). (Two notes: While the hyphen is allowed in table names, it is also a reserved character in SQL, so if you use it you must remember to double quote the table name in your queries. Using a double-underscore (‘\_\__’) is not allowed as it is reserved for other features within Pinot.)                                                     |
```

**Review by matthewhelmke — CHANGES_REQUESTED** [review]
I appreciate very much that you submitted this PR! Instead of rejecting the PR based on the comment above, let's modify it to help prevent others from having the same problem.

**Comment by jflayhart** [comment]
love the suggestion and makes perfect sense thanks! much clearer.

**Review by matthewhelmke — APPROVED** [review]
LGTM, thank you!

**Pull request merged**